tweet_editor.py

```diff
@@ -0,0 +1,258 @@
+                group_not_found = False
+            group_list_index +=1
+        items_in_each_group[group_list_index-2] += 1
+    print(items_in_each_group)
+    print(sentiment_averages)
+    for index in range(len(sentiment_averages)):
+        if items_in_each_group[index] != 0:
+            print(items_in_each_group[index])
+            averaged_value = float(sentiment_averages[index])/float(items_in_each_group[index])
+            sentiment_averages[index] = averaged_value
+            print(averaged_value)
+    sentiment_averages
+    return sentiment_averages
+
+def plot_tweet(dates, tweet_sentiment_list):
+    '''
+    This function plots the data in a scatter plot. The data is plotted in
+    the form of positive - negative score for each tweet. This gives a
+    holistic score for each tweet.
+    '''
+    #list of trumps standing over times
+    political_standing_date = [191, 196, 201, 206, 211, 216, 221, 226, 231, 236, 241, 246, 251, 256, 261, 266]
+    political_standing = [42.5, 41.5, 43, 46.5, 46, 43.5, 42.5, 40.5, 43.5, 42.5, 44, 43, 42.5, 46, 45, 45.5]
+    for number in range(len(political_standing)):
+        political_standing[number] = political_standing[number]/60-.65
+    positive_minus_negative_list = []
+    #splits the list of sentiments up into positive and negative
+    positivity_list = tweet_sentiment_list[0]
+    negativity_list = tweet_sentiment_list[1]
+    #the for loop makes a list of the positive values minus the negative values
+    for items in range(len(tweet_sentiment_list[0])):
+        positive_minus_negative_value = positivity_list[items]- negativity_list[items]
+        positive_minus_negative_list.append(positive_minus_negative_value)
+    #makes a scatter plot with dates on the x axis and the positive minus
+    #the negative on the y. It also adds labels and axes.
+    plt.plot(dates[11:len(dates)-8], positive_minus_negative_list[11:len(dates)-8], label = 'Positivity')
+    plt.plot(political_standing_date, political_standing, color ='r', label = 'Political Standing')
+    #plt.scatter(dates, tweet_sentiment_list[1], label = 'Negativity', color = 'r')
+    plt.title('Average Tweet Sentiment vs. Political Standing')
+    plt.xlabel('Date')
+    plt.ylabel('Positivity-Negativity')
+    plt.legend()
+    plt.show()
+
+
+def access_twitter_files ():
+    '''
+    This is the main function. It loads the tweets from a file so they can be
+    edited without accessing twitter again. It then parses out the tweets and
+    date from the other information. The data is then put through the
+    other functions in the file, which analyze the data and graph it.
+    '''
+    #This opens the file that the other atom file created when it pulled the
+    #tweets from twitter
+    files = open("trump_tweets.txt", 'rb')
+    tweets = pickle.load(files)
+    files.close
+
+    #this is the contents of the file which contains all the tweets, dates, etc.
+    tweets = str(tweets)
```

Comment:
I may be wrong about this, but I think a lot of the string manipulation could be avoided if you had just kept the tweets as a list? I think each tweet is an object that you can directly access.